crates/common_types/src/payments.rs

```
@@ -77,6 +81,64 @@ impl AuthenticationConnectorAccountMap {
     }
 }
 
+/// A wrapper type for merchant country codes that provides validation and conversion functionality.
+///
+/// This type stores a country code as a string and provides methods to validate it
+/// and convert it to a `Country` enum variant.
+#[derive(
+    Serialize, Deserialize, Debug, Clone, PartialEq, Eq, FromSqlRow, AsExpression, ToSchema,
+)]
+#[diesel(sql_type = Text)]
+#[serde(deny_unknown_fields)]
+pub struct MerchantCountryCode(String);
+
+impl MerchantCountryCode {
+    /// Returns the country code as a string.
+    pub fn get_country_code(&self) -> String {
+        self.0.clone()
+    }
+
+    /// Validates the country code and returns a `Country` enum variant.
+    ///
+    /// This method attempts to parse the country code as a u32 and convert it to a `Country` enum variant.
+    /// If the country code is invalid, it returns a `ValidationError` with the appropriate error message.
+    pub fn validate_and_get_country_from_merchant_country_code(
+        &self,
+    ) -> errors::CustomResult<common_enums::Country, errors::ValidationError> {
+        let country_code = self.get_country_code();
+        let code = country_code
+            .parse::<u32>()
+            .map_err(|_| errors::ValidationError::IncorrectValueProvided {
+                field_name: "merchant_country_code",
+            })
```

why are you ignoring the error? can we do change_context instead?